lib/maplibre.ex

```diff
@@ -526,4 +595,78 @@ defmodule MapLibre do
   defp to_style(%{}), do: %{"version" => 8}
   defp to_style(style) when is_map(style), do: style
   defp to_style(style), do: Jason.decode!(style)
+
+  defp geometry_from_table(data, spec, []) do
+    geometries =
+      data
+      |> points(spec)
+      |> Enum.map(&%Geo.Point{coordinates: &1})
+
+    Geo.JSON.encode!(%Geo.GeometryCollection{geometries: geometries}, feature: true)
+  end
+
+  defp geometry_from_table(data, spec, properties) do
+    points = points(data, spec)
+    properties = properties(data, properties)
+    geometries = Enum.zip(points, properties)
+
+    geometries =
+      for {point, props} <- geometries, do: %Geo.Point{coordinates: point, properties: props}
+
+    Geo.JSON.encode!(%Geo.GeometryCollection{geometries: geometries}, feature: true)
+  end
+
+  def points(data, {format, [lng, lat]}) do
+    validate_columns!(data, [lng, lat])
+
+    data
+    |> Table.to_columns(only: [lng, lat])
+    |> then(&Enum.zip(&1[lng], &1[lat]))
+    |> Enum.map(&parse_coordinates(&1, format))
+  end
+
+  def points(data, {format, coordinates}) do
+    validate_columns!(data, [coordinates])
+
+    data
+    |> Table.to_columns(only: [coordinates])
+    |> Map.get(coordinates)
+    |> Enum.map(&parse_coordinates(&1, format))
+  end
+
+  defp properties(data, properties) do
+    validate_columns!(data, properties)
+
+    data
+    |> Table.to_rows(only: properties)
+    |> Enum.to_list()
+  end
+
+  defp parse_coordinates({lng, lat}, :lng_lat), do: {lng, lat}
+  defp parse_coordinates({lng, lat}, :lat_lng), do: {lat, lng}
+
+  defp parse_coordinates(coordinates, format) do
+    Regex.named_captures(~r/(?<lng>-?\d+\.?\d*)\s*[,;\s]\s*(?<lat>-?\d+\.?\d*)/, coordinates)
+    |> case do
+      %{"lat" => lat, "lng" => lng} -> if format == :lng_lat, do: {lng, lat}, else: {lat, lng}
+      _ -> raise ArgumentError, "unsupported coordinates data"
+    end
+  end
+
+  defp validate_columns!(data, columns) do
+    data_columns = columns_for(data)
+    missing_column = Enum.find(columns, &(&1 not in data_columns))
+
+    if missing_column,
+      do: raise(ArgumentError, "column #{inspect(missing_column)} was not found.")
+  end
+
+  defp columns_for(data) do
+    with {_, %{columns: columns}, _} <- Table.Reader.init(data),
+         true <- Enum.all?(columns, &String.Chars.impl_for/1) do
+      Enum.map(columns, &to_string/1)
```

Comment:
I don't think we should rule out the table if it has weird columns. What do you think about this:

```suggestion
    with {_, %{columns: columns}, _} <- Table.Reader.init(data) do
      for column <- columns do
        if is_atom(column), do: Atom.to_string(column), else: column
      end
```